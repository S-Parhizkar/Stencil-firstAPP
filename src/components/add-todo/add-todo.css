:host {
  display: block;
}

button {

    background-color: #0078d0;
    border: 0;
    border-radius: 30px;
    color: #fff;
    cursor: pointer;
    display: inline-block;
    font-family: system-ui,-apple-system,system-ui,"Segoe UI",Roboto,Ubuntu,"Helvetica Neue",sans-serif;
    font-weight: 600;
    outline: 0;
    padding: 0.5em 1em;
    position: relative;
    text-align: center;
    text-decoration: none;
    transition: all .3s;
    user-select: none;
    -webkit-user-select: none;
    touch-action: manipulation;
    margin: 0 0.25em;
    font-family: cursive;
  }
  button:hover {
    box-shadow: rgba(255, 255, 255, .2) 0 3px 15px inset, rgba(0, 0, 0, .1) 0 3px 5px, rgba(0, 0, 0, .1) 0 10px 13px;
    transform: scale(1);
  }

.btn:disabled {
  opacity: 0.5;
}

span {
  color: var(--primary-70);
  font-weight: bold;
  font-size: 20px;
  margin: 0 0.25em;
  height: 30px;
  padding: 0.2em 0.5em;
}

button:disabled {
  opacity: 0.5;
}

input {
  -webkit-border-radius: 5px;
  -moz-border-radius: 5px;
  border-radius: 5px;
  font-family: cursive;
  border: 1px solid #848484;
  outline: 0;
  height: 30px;
  width: 80%;
}
