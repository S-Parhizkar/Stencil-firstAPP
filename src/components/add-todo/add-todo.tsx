import { Component, h, Event, EventEmitter } from '@stencil/core';
// import { ITodo } from '../types';


@Component({
  tag: 'add-todo',
  styleUrl: 'add-todo.css',
  shadow: true,
})
export class AddTodo {
  private $textInput?: HTMLInputElement;

  //Post/add to do
  @Event({ eventName: 'add-more' }) sendTodoHandler:EventEmitter<string>;

  handleAddToDo() {
    const inputValue = this.$textInput.value;
    this.sendTodoHandler.emit(inputValue);

    console.log('3- test add-todo side ', inputValue);
    this.emptyinputValue();
  }
  emptyinputValue(){
    this.$textInput.value = "";
    }
  
  render() {
    return (

  <div class='container '>
          <input class={"inputTodo"} type="text"  placeholder='   Add your new " to do " to this list here' 
          ref={el => this.$textInput = el as HTMLInputElement} 
          />
          <button onClick={() => this.handleAddToDo()}> add to do</button>
  </div>
     
    );
  }
}
