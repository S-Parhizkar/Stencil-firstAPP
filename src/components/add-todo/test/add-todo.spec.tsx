import { newSpecPage, SpecPage } from '@stencil/core/testing';
import { h } from '@stencil/core';
import { AddTodo } from '../add-todo';


describe('add-more', () => {
  const todo: ITodo = {
    completed: true,
    id: 2,
    order: 3,
    title: 'sport',
    url: 'https://my-list-to-do.netlify.app/',
  };

  let page: SpecPage;

  beforeEach(async ()=> {
    page = await newSpecPage({
      components: [AddTodo],
      template: () => <add-todo todo={todo}></add-todo>,
    });
    await page.waitForChanges();
  })

it('render', ()=>{
  expect(page.root).toMatchSnapshot();
})


it('should emit [add-more] event to add new todo once clicked on Add to do button', async () => {
  const myMock = jest.fn();
  page.body.addEventListener('add-more', myMock);
  (page.root.shadowRoot.querySelector('input.inputTodo') as HTMLInputElement).value = "sport" as string;
  (page.root.shadowRoot.querySelector('button') as HTMLInputElement).click();
  await page.waitForChanges();
  const mockTodo = { ...todo };
    mockTodo.title = todo.title.toUpperCase().trim();
    expect(myMock).toHaveBeenCalledWith(expect.objectContaining({detail:todo}));
})
 
});
