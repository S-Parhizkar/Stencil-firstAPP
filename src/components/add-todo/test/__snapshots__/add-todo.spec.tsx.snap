// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add-more render 1`] = `
<add-todo>
  <mock:shadow-root>
    <div class="container">
      <input class="inputTodo" placeholder="   Add your new &quot; to do &quot; to this list here" type="text">
      <button>
        Add to do
      </button>
    </div>
  </mock:shadow-root>
</add-todo>
`;
